lib/mrsk/configuration.rb

```diff
@@ -252,10 +252,11 @@ def role_names
 
     def git_version
       @git_version ||=
-        if system("git rev-parse")
-          uncommitted_suffix = Mrsk::Utils.uncommitted_changes.present? ? "_uncommitted_#{SecureRandom.hex(8)}" : ""
+        if system("git describe")
+          git_reference = `git describe --always`
+          dirty_suffix  = "dirty-#{SecureRandom.hex(8)}" if Mrsk::Utils.uncommitted_changes.present?
```

Comment:
I suspect the idea of including the "_uncommited_" or "-dirty-" part in the version was originally intended to be a hint to the developer that their was uncommited changes in their build.

I am wondering if it would be better to just print a warning to the console and keep the git_version a bit more succinct? We would still need the random chars to avoid image cache clashes though.